src/com/TTT/MiniMax.java

```diff
@@ -5,21 +5,94 @@
 
 public class MiniMax {
 
-    public int nextMove(Board board, String currentPlayerSymbol){
-        return -1;
+    private PlayerMarkers maxPlayer;
+
+    public int nextMove(Board board, PlayerMarkers maxPlayer){
+        this.maxPlayer = maxPlayer;
+
+        List<Integer> moves = board.availableMoves();
+        List<Integer> scores = new ArrayList<>();
+
+        for(int move : moves) {
+            Board clone = board.clone();
+            clone.placeMarker(move, maxPlayer.symbol());
+            scores.add(miniMax(maxPlayer, clone));
+        }
+
+        int bestScore = -200;
+        int bestMove = -1;
+        for(int i = 0; i < scores.size(); i++) {
+            if(scores.get(i) > bestScore) {
+                bestScore = scores.get(i);
+                bestMove = moves.get(i);
+            }
+        }
+        return bestMove;
     }
 
-    public int getScoreForMove(int move, Board board, String playerSymbol, String maxingPlayerSymbol, String minimizingPlayerSymbol, int depth) {
+    public int miniMax(PlayerMarkers currentPlayer, Board board) {
+        BoardEvaluator evaluator = new BoardEvaluator(board);
+        if(evaluator.evaluate().equals(GameState.Win) && currentPlayer == maxPlayer) {
+            return 10;
+        }
+
+        if(evaluator.evaluate().equals(GameState.Win) && currentPlayer != maxPlayer) {
+            return -10;
+        }
+
+        if(evaluator.evaluate().equals(GameState.Tie)) {
+            return 0;
+        }
+
+        List<Integer> scores = new ArrayList<>();
 
+        currentPlayer = changePlayer(currentPlayer);
```

Comment:
This was the line that made it work. Previously I was changing the player when making the recursive call to minimax.